snooty/rstparser.py

```diff
@@ -0,0 +1,392 @@
+import docutils.frontend
+import docutils.nodes
+import docutils.parsers.rst
+import docutils.parsers.rst.directives
+import docutils.parsers.rst.roles
+import docutils.parsers.rst.states
+import docutils.statemachine
+import docutils.utils
+import re
+from dataclasses import dataclass
+from pathlib import Path, PurePath
+from typing import Any, Callable, Dict, Generic, Optional, List, Tuple, \
+    Type, TypeVar, Iterable, Sequence
+from typing_extensions import Protocol
+from .gizaparser.parse import load_yaml
+from .gizaparser import nodes
+from .types import Diagnostic, ProjectConfig
+from .flutter import checked, check_type, LoadError
+
+PAT_EXPLICIT_TILE = re.compile(r'^(?P<label>.+?)\s*(?<!\x00)<(?P<target>.*?)>$', re.DOTALL)
```

Comment:
Could you explain how this regex works?  I'm especially not sure what `(?<!\x00)` does.  Does the mean "not followed by a null byte"?  I thought the null byte would never be considered part of the string.

Reply:
No idea; it was taken verbatim from docutils. I'll add a comment to that effect :+1: